ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py

```diff
@@ -780,6 +780,22 @@ def array_values(
                         exclude_max=exclude_max,
                     ),
                 )
+                if "float" in dtype or "complex" in dtype:
```

Comment:
I'm not quite sure why we're doing these checks once again. They're done earlier in the function

Reply:
I implented the checks cause on running the function nancumprod, it rendered the error, floats function does not allow the 'allow_subnormal' argument that's why i Had to implement the checks